Breeding-Insight/bi-web
# [BI-1199] - Choose file format for germplasm export

# Description
**Story:** [BI-1199 - Choose file format for germplasm export](https://breedinginsight.atlassian.net/browse/BI-1199)

Changes to frontend to enable modal selection of file type when downloading germplasm list and passing of user selection to backend.

Added new reusable modal component SelectModal that takes in title and a list of options to display, and utilized it to display file types upon user clicking download and pass that information on click.

# Dependencies
[bi-api/BI-1199](https://github.com/Breeding-Insight/bi-api/pull/193)

# Testing
- Open Germplasm List
- Click Download
- Verify modal pops up and shows
- Verify each file type can be selected
- Verify when cancel clicked the modal closes and nothing is downloaded
- Verify for each file type (csv, xls, xlsx) that when it is selected and download is clicked the file is downloaded with the proper extension and the correct data

# Checklist:

- [X] I have run TAF: [_\<link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/2545220951)
